docs/graphx-programming-guide.md

```diff
@@ -708,7 +708,9 @@ messages remaining.
 > messaging function.  These constraints allow additional optimization within GraphX.
 
 The following is the type signature of the [Pregel operator][GraphOps.pregel] as well as a *sketch*
-of its implementation (note calls to graph.cache have been removed):
+of its implementation (note: to avoid stackOverflowError due to long lineage chains, graph and 
```

Comment:
> I guess it would help reviewers to understand if you could explain those changes in the impl sketch is required to use checkpoint

Sorry @felixcheung, I don't understand what you mean here. Can you elaborate?